### PR TITLE
feat: shell support cloud

### PR DESCRIPTION
**Pull request by zhaoyanggh**

### CODE REVIEW

**Review by afwerar — APPROVED** [review]
LGTM

File: tools/shell/inc/shellInt.h [hunk 1]
```diff
@@ -62,11 +67,18 @@ typedef struct {
   bool        is_check;
   bool        is_startup;
   bool        is_help;
+#ifdef WEBSOCKET
+  bool        restful;
+  bool        cloud;
+#endif
```
Comment (sangshuduo) [inline tools/shell/inc/shellInt.h]: how about combining ws arguments to one block for easerily maintenance?

File: tools/shell/inc/shellInt.h [hunk 2]
```diff
@@ -53,6 +57,7 @@ typedef struct {
   const char* cfgdir;
   const char* commands;
   const char* netrole;
+  char* dsn;
```
Comment (sangshuduo) [inline tools/shell/inc/shellInt.h]: dsn can be used for non-ws mode?

File: tools/shell/inc/shellInt.h [hunk 3]
```diff
@@ -83,6 +95,10 @@ typedef struct {
   SShellHistory   history;
   SShellOsDetails info;
   TAOS*           conn;
+#ifdef WEBSOCKET
+  WS_TAOS*        ws_conn;
+  bool		      stop_query;
```
Comment (sangshuduo) [inline tools/shell/inc/shellInt.h:100]: stop_query need be align with others

**Review by sangshuduo — APPROVED** [review]
LGTM with a few suggestions

**Pull request merged**